src/magic_folder/config.py

```diff
@@ -0,0 +1,419 @@
+"""
+Configuration and state database interaction.
+
+See also docs/config.rst
+"""
+
+from os import (
+    mkdir,
+    urandom,
+)
+from base64 import (
+    urlsafe_b64encode,
+)
+
+import attr
+
+import sqlite3
+
+from nacl.signing import (
+    SigningKey,
+)
+from nacl.encoding import (
+    Base32Encoder,
+)
+
+from twisted.internet.endpoints import (
+    serverFromString,
+)
+from twisted.python.filepath import (
+    FilePath,
+)
+
+from .snapshot import (
+    LocalAuthor,
+)
+
+
+class ConfigurationError(Exception):
+    """
+    The configuration is unusable for some reason
+    """
+
+
+_global_config_version = 1
+
+_global_config_schema = """
+CREATE TABLE version
+(
+    version INTEGER  -- contains one row, set to 1
+);
+
+CREATE TABLE magic_folders
+(
+    name          TEXT PRIMARY KEY,  -- UTF8 name of this folder
+    location      TEXT               -- UTF8 path to this folder's configuration/state
+);
```

Comment:
I *think* UTF8 bytes is the right choice here?  I'm not sure if there are some nasty surprises waiting on macOS or Windows though.  I think it wouldn't hurt to be explicit about these being bytes and not unicode (though "UTF8" pretty strongly implies bytes).  SQLite3 "type affinity" is a bit odd.  This is a Python 3 transcript but the basic idea holds for Python 2.  Notice the types of the 3 returned rows.

```
>>> import sqlite3
>>> conn = sqlite3.connect(":memory:")
>>> curs = conn.cursor()
>>> curs.execute("CREATE TABLE foo ( a TEXT )")
<sqlite3.Cursor object at 0x7f0b9f771d50>
>>> curs.execute("INSERT INTO foo VALUES (?)", (u"\N{SNOWMAN}",))
<sqlite3.Cursor object at 0x7f0b9f771d50>
>>> curs.execute("INSERT INTO foo VALUES (?)", (u"snowman",))
<sqlite3.Cursor object at 0x7f0b9f771d50>
>>> curs.execute("INSERT INTO foo VALUES (?)", (b"snowman",))
<sqlite3.Cursor object at 0x7f0b9f771d50>
>>> curs.execute("SELECT * FROM foo")
<sqlite3.Cursor object at 0x7f0b9f771d50>
>>> curs.fetchall()
[('☃',), ('snowman',), (b'snowman',)]
>>> 
```

Reply:
Mostly I'm looking at other SQLite3 code for inspiration; if there's something besides `TEXT` that we should use here (or some other comment / `.encode()` calls) I'm all ears :)